File: examples/sdks/python/execute.py
import os
import uuid

from iroute import IRouteClient

client = IRouteClient(
    os.getenv("IROUTE_URL", "http://localhost:8080"),
    token=os.getenv("IROUTE_TOKEN"),
    tenant_id=os.getenv("IROUTE_TENANT", "demo"),
    actor_id=os.getenv("IROUTE_ACTOR", "sdk-example"),
)
result = client.execute({
    "taskType": "email.draft",
    "input": {"purpose": "Confirm the SDK quick start."},
    "idempotencyKey": f"python-example-{uuid.uuid4()}",
})
print(result)

File: examples/sdks/node/execute.mjs
import { randomUUID } from 'node:crypto';
import { IRouteClient } from '../../../sdks/node/dist/index.js';

const client = new IRouteClient(
  new URL(process.env.IROUTE_URL ?? 'http://localhost:8080'),
  {
    token: process.env.IROUTE_TOKEN,
    tenantId: process.env.IROUTE_TENANT ?? 'demo',
    actorId: process.env.IROUTE_ACTOR ?? 'sdk-example'
  }
);
const result = await client.execute({
  taskType: 'email.draft',
  input: { purpose: 'Confirm the SDK quick start.' },
  idempotencyKey: `node-example-${randomUUID()}`
});
console.log(JSON.stringify(result, null, 2));
